src/components/DocumentCard/DocumentCard.tsx

```diff
@@ -31,16 +27,30 @@ export class DocumentCard extends React.Component<IDocumentCardProps, any> {
           css(
             'ms-DocumentCard',
             {
-              'ms-DocumentCard--actionable': onClick ? true : false,
+              'ms-DocumentCard--actionable': actionable,
               'ms-DocumentCard--compact': type === DocumentCardType.compact ? true : false
             },
             className
           )
         }
-        onClick={ onClick }
+        onClick={ actionable ? this._onClick : null }
         style={ style }>
         { children }
       </div>
     );
   }
+
+  @autobind
+  private _onClick(ev: React.MouseEvent): void {
+    let { onClick, onClickHref } = this.props;
+
+    if (onClick) {
+      onClick(ev);
+    } else if (!onClick && onClickHref) {
+      // If no onClick Function was provided and we do have an onClickHref, redirect to the onClickHref
+      window.location.href = onClickHref;
```

Comment:
A better approach would be  to create an anchor layer that is hosted within the div, and add a sibling to that anchor which overlays it, and contains the doc card stuff. That way, you aren't nesting anchors, but you can still make the card behave as an anchor with all the shift/ctrl click/status-bar-on-hover anchory goodness.